### PR TITLE
Saving the progress report PDF

### PR DESCRIPTION
**Pull request by Eb-Zeero**
This change is [Reviewable](https://reviewable.io/reviews/saltastroops/salt-api/267)

### CODE REVIEW

File: saltapi/service/proposal_service.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import pathlib
 import urllib.parse
 from io import BytesIO
-from typing import Any, Dict, List, Optional, cast
+from typing import Any, Dict, List, Optional, cast, Tuple
```
Comment (hettlage) [inline saltapi/service/proposal_service.py]: The imports from `typing` aren't sorted properly.

File: saltapi/service/proposal_service.py [hunk 2]
```diff
@@ -189,52 +189,91 @@ async def put_proposal_progress(
...
+    @staticmethod
+    async def save_additional_pdf(
+            proposal_code: str,
+            additional_pdf_filename: str,
+            additional_pdf_content: bytes
+    ) -> None:
+        additional_pdf_path = ProposalService._included_dir(proposal_code) / additional_pdf_filename
+        additional_pdf_path.write_bytes(additional_pdf_content)
+
+    @staticmethod
```
Comment (hettlage) [inline saltapi/service/proposal_service.py]: This method is identical to `save_additional_pdf`, isn't it?

**Review by hettlage — CHANGES_REQUESTED** [review]

**Review by Eb-Zeero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api/267)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @hettlage)

___
*[`saltapi/service/proposal_service.py` line 4 at r1](https://reviewable.io/reviews/saltastroops/salt-api/267#gh-1500724700:-NrZC5BcDJraUrRxxOit:b-896fix) ([raw file](https://github.com/saltastroops/salt-api/blob/e6dccc2e3710cf9d0913619ce4023047b348dd48/saltapi/service/proposal_service.py#L4)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

The imports from `typing` aren't sorted properly.
</blockquote></details>

Done.
___
*[`saltapi/service/proposal_service.py` line 269 at r1](https://reviewable.io/reviews/saltastroops/salt-api/267#gh-1500743511:-NrZD9YV9JVkA17J5jcX:b-896fix) ([raw file](https://github.com/saltastroops/salt-api/blob/e6dccc2e3710cf9d0913619ce4023047b348dd48/saltapi/service/proposal_service.py#L269)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

This method is identical to `save_additional_pdf`, isn't it?
</blockquote></details>

Done.

File: saltapi/service/proposal_service.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import pathlib
 import urllib.parse
 from io import BytesIO
-from typing import Any, Dict, List, Optional, cast
+from typing import Any, cast, Dict, List, Optional, Tuple
```
Comment (hettlage) [inline saltapi/service/proposal_service.py]: This is not sorted correctly according to Ruff; `cast` must come last.

File: saltapi/service/proposal_service.py [hunk 4]
```diff
@@ -189,52 +189,82 @@ async def put_proposal_progress(
...
 
-        # Save the additional file to disk. This is done last as the file should not be
-        # stored if the progress report cannot be stored in the database.
-        if additional_pdf_filename:
-            additional_pdf_path = (
-                ProposalService._included_dir(proposal_code) / additional_pdf_filename
-            )
-            additional_pdf_path.write_bytes(additional_pdf_content)
+    @staticmethod
+    async def save_progress_report_pdf(
```
Comment (hettlage) [inline saltapi/service/proposal_service.py]: Looking at the code, it doesn't really matter whether the file is a pdf, so you might remove `pdf` from the function name and from the variable names within the function.

**Review by hettlage — CHANGES_REQUESTED** [review]

**Review by Eb-Zeero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api/267)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @hettlage)

___
*[`saltapi/service/proposal_service.py` line 4 at r2](https://reviewable.io/reviews/saltastroops/salt-api/267#gh-1502141565:-NrZQM9J9_epIeuyTtl-:b-896fix) ([raw file](https://github.com/saltastroops/salt-api/blob/dbdda03b1ce5183c7286d9a0feb512a61632cfdd/saltapi/service/proposal_service.py#L4)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

This is not sorted correctly according to Ruff; `cast` must come last.
</blockquote></details>

Done.
___
*[`saltapi/service/proposal_service.py` line 261 at r2](https://reviewable.io/reviews/saltastroops/salt-api/267#gh-1502143426:-NrZQMzU10vk_Eu8GVwa:b-896fix) ([raw file](https://github.com/saltastroops/salt-api/blob/dbdda03b1ce5183c7286d9a0feb512a61632cfdd/saltapi/service/proposal_service.py#L261)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Looking at the code, it doesn't really matter whether the file is a pdf, so you might remove `pdf` from the function name and from the variable names within the function.
</blockquote></details>

Done.

File: saltapi/service/proposal_service.py [hunk 5]
```diff
@@ -189,52 +189,82 @@ async def put_proposal_progress(
...
 
-        # Save the additional file to disk. This is done last as the file should not be
-        # stored if the progress report cannot be stored in the database.
-        if additional_pdf_filename:
-            additional_pdf_path = (
-                ProposalService._included_dir(proposal_code) / additional_pdf_filename
-            )
-            additional_pdf_path.write_bytes(additional_pdf_content)
+    @staticmethod
+    async def save_progress_report(
```
Comment (hettlage) [inline saltapi/service/proposal_service.py]: Thanks for dropping the `_pdf`. But now there is another issue: The name `save_progress_report` indicates that you save the progress report. Calling the function `save_progress_report_file` thus might be slightly better.

**Review by hettlage — CHANGES_REQUESTED** [review]

**Review by Eb-Zeero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api/267)* status: 0 of 1 files reviewed, 5 unresolved discussions (waiting on @hettlage)

___
*[`saltapi/service/proposal_service.py` line 261 at r3](https://reviewable.io/reviews/saltastroops/salt-api/267#gh-1502355271:-Nr_CTlP6LsKRquC95w_:b-896fix) ([raw file](https://github.com/saltastroops/salt-api/blob/cd7858a824d55c4c6c04d066c67423593b9ad599/saltapi/service/proposal_service.py#L261)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Thanks for dropping the `_pdf`. But now there is another issue: The name `save_progress_report` indicates that you save the progress report. Calling the function `save_progress_report_file` thus might be slightly better.
</blockquote></details>

Done.

**Review by hettlage — APPROVED** [review]

**Pull request merged**